docs/docs/guides/upgrade-to-v5.md

```diff
@@ -0,0 +1,265 @@
++ export { auth as default } from "./auth"
+```
+
+For advanced use cases, you can use `auth` as a wrapper for your Middleware:
+
+```ts title="middleware.ts"
+import { auth } from "./auth"
+
+export default auth(req => {
+  // req.auth
+})
+```
+
+Read the [Middleware docs](/reference/nextjs#in-middleware) for more details.
+
+### Server Component
+
+NextAuth.js v4 has supported reading the session in Server Components for a while via `getServerSession`. This has been also simplified to the same `auth()` function.
+
+```diff title="app/page.tsx"
+- import { authOptions } from 'pages/api/auth/[...nextauth]'
+- import { getServerSession } from "next-auth/next"
++ import { auth } from "../auth"
+
+export default async function Page() {
+-  const session = await getServerSession(authOptions)
++  const session = await auth()
+  return (<p>Welcome {session?.user.name}!</p>)
+}
+```
+
+### Client Component
+
+Imports from `next-auth/react` are now marked with the `"use client"` directive. [Read more](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive).
+
+If you have previously used `getSession()` or other imports server-side, you'll have to change it to use the [`auth()`](/reference/nextjs#auth) method instead.
+
+`getCsrfToken` and `getProviders` are still available as imports, but we plan to deprecate them in the future and introduce a new API to get this data server-side.
+
+Client side: Instead of using these APIs, you can make a fetch request to the `/api/auth/providers` and `/api/auth/csrf` endpoints respectively.
+
+Server-side: TBD
+
+## Adapters
+
+### Adapter type
+
+The `Adapter` type is uncommon, unless you are writing your own adapter. If you are using an existing adapter, you can likely ignore this change.
+
+Otherwise, the `Adapter` type can now be imported from `@auth/core/adapters` instead of `next-auth/adapters`.
+
+```diff title="pages/api/auth/[...nextauth].ts"
+- import type { Adapter } from "next-auth/adapters"
++ import type { Adapter } from "@auth/core/adapters"
+```
+
+This is because the - previously - NextAuth.js scoped [`@next-auth/*-adapter`](https://www.npmjs.com/search?q=next-auth%20adapter) database adapters should work in any framework the same way (there is nothing Next.js specific in these adapters), so it was unnecessary to keep this type tied to the `next-auth` package.
+
+### Database Migrations
+
+NextAuth.js v5 does not introduce any breaking changes to the database schema. However, since OAuth 1.0 support is dropped, the already optional `oauth_token_secret` and `oauth_token` fields can be removed from the `account` table, if you are not using them.
+
+Furthermore, previously uncommon fields like GitHub's `refresh_token_expires_in` fields were required to be added to the `account` table. This is no longer the case, and you can remove it if you are not using it. if you do, make sure to return it via the new [`acconut()` callback](/reference/core/providers#account)
```

Comment:
```suggestion
Furthermore, previously uncommon fields like GitHub's `refresh_token_expires_in` fields were required to be added to the `account` table. This is no longer the case, and you can remove it if you are not using it. if you do, make sure to return it via the new [`account()` callback](/reference/core/providers#account)
```

Reply:
Also, https://auth-docs-git-feat-nextjs-auth-authjs.vercel.app/reference/core/providers#account is 404ing